lib/utilities/reportHandler.js

```diff
@@ -22,6 +22,26 @@ function isAuthorized(clientIP, req) {
         req.headers['x-scal-report-token'] === config.reportToken;
 }
 
+const itemScanRefreshDelay = 1000 * 30 * 60; // 30 minutes
+let lastItemScanTime = undefined;
+let lastItemScanResult = undefined;
+
+function countItems(log, cb) {
+    // The value is cached, as scanning all the dataset can be I/O intensive
+    if (lastItemScanTime === undefined ||
+        (Date.now() - lastItemScanTime) > itemScanRefreshDelay) {
+        return metadata.countItems(log, (err, res) => {
+            if (err) {
+                return cb(err);
+            }
+            lastItemScanTime = Date.now();
+            lastItemScanResult = res;
+            return cb(null, res);
+        });
+    }
+    return cb(null, lastItemScanResult);
```

Comment:
I would suggest wrapping this in `process.nextTick` to avoid any chance of [releasing zalgo](http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony)